presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java

```diff
@@ -376,6 +379,9 @@ private void load(boolean recursive)
             }
 
             block = requireNonNull(loader.load(), "loader returned null");
+            if (block.getPositionCount() != positionsCount) {
+                throw new IllegalStateException(format("Loaded block positions count (%s) doesn't match lazy block positions count (%s)", block.getPositionCount(), positionsCount));
```

Comment:
Maybe say "block is invalid" or something about it being not loadable because there is another projection... if this ever appears it will be very confusing

Reply:
> Maybe say "block is invalid" or something about it being not loadable because there is another projection

this only validates that loaded block is of same size as lazy block.